tests/model_forms/tests.py

```diff
@@ -3069,3 +3069,19 @@ def test_setattr_raises_validation_error_non_field(self):
             '__all__': ['Cannot set attribute'],
             'title': ['This field cannot be blank.']
         })
+
+class ModelFormArrayFieldTests(TestCase):
+
+    def setUp(self):
+        self.form_class = modelform_factory(
+            model=ArrayFieldTestModel,
+            fields=['test_field']
+        )
+
+    def test_array_field_with_multiple_fields(self):
+        form = self.form_class(data={'test_field': '1,2'}, files=None)
```

Comment:
I removed the invalid case since it's already passing without the patch. We could add it in a separate commit but I'm not sure it's needed. If we did add it, I'd like if it check the error message similar to other tests, e.g. 
```python
errors = cm.exception.error_list
self.assertEqual(len(errors), 2)
first_error = errors[0]
self.assertEqual(
    first_error.message,
    'Item 0 in the array did not validate: Select a valid choice. 3 is not one of the available choices.'
)
second_error = errors[1]
self.assertEqual(
    second_error.message,
    'Item 1 in the array did not validate: Select a valid choice. 4 is not one of the available choices.'
)
```